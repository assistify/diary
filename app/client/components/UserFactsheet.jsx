import React from 'react';
import { PropTypes } from 'prop-types';
import Media from 'react-bulma-components/lib/components/media';
import Image from 'react-bulma-components/lib/components/image';
import Heading from 'react-bulma-components/lib/components/heading';
import Container from 'react-bulma-components/lib/components/container';
import User from './User';
import { TeamContext } from '../lib/teamContext';


export default function UserFactsheet(props) {
  const { username, children } = props;
  return (
    <TeamContext.Consumer>
      {teamContext => (
        <Media>
          <Container>
            <Media.Item renderAs="figure" position="left">
              <Image size={128} className="avatar" alt={username} src={`${teamContext.serverUrl}/avatar/${username}`} fallback="http://bulma.io/images/placeholders/128x128.png" />
            </Media.Item>
          </Container>

          <Container style={{ margin: 'auto' }}>
            <Media.Item>
              <Heading size={5}>
                <User username={username} />
              </Heading>
              <Container>{children}</Container>
            </Media.Item>
          </Container>
        </Media>
      )}
    </TeamContext.Consumer>
  );
}

UserFactsheet.defaultProps = {
  children: []
};

UserFactsheet.propTypes = {
  username: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};
