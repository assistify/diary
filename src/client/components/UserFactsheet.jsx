import React from 'react';
import { PropTypes } from 'prop-types';
import Media from 'react-bulma-components/lib/components/media';
import Heading from 'react-bulma-components/lib/components/heading';
import User from './User';
import Avatar from './Avatar';
import '../styles/components/UserFactSheet.scss';

export default class UserFactsheet extends React.Component {  // eslint-disable-line
  constructor(props) {
    super(props);
    this.state = {
      popupVisible: false
    };
  }

  showConverter() {
    this.setState({ popupVisible: true });
  }

  render() {
    const {
      member,
      children,
      contentEditable,
      updateValue
    } = this.props;

    const statusKnownCheckbox = contentEditable && (
      <div className="statusKnownCheckbox">
        <input
          type="checkbox"
          checked={member.statusKnown}
          onChange={e => updateValue(member.username, 'statusKnown', e.target.checked)}
        />
        &nbsp;provided info
      </div>
    );

    const { popupVisible } = this.state;
    const popup = popupVisible && (
      <div className="popup">
        <textarea>{JSON.stringify(member, null, 2)}</textarea>
        <button type="button" onClick={() => this.setState({ popupVisible: false })}>Ok</button>
      </div>
    );

    const converterButton = member.statusKnown && (
      <button type="button" className="converterButton" onClick={() => this.showConverter()}>âœŽ</button>
    );

    return (
      <Media>
        <Media.Item renderAs="figure" position="left">
          <Avatar username={member.username} />
        </Media.Item>

        <div className="factSheetInfo">
          { converterButton }
          <Heading size={5}>
            <User username={member.username} />
          </Heading>
          {statusKnownCheckbox}
          {children}
        </div>
        {popup}
      </Media>
    );
  }
}

UserFactsheet.defaultProps = {
  children: []
};

UserFactsheet.propTypes = {
  member: PropTypes.shape({
    username: PropTypes.string.isRequired,
    statusKnown: PropTypes.bool.isRequired,
    past: PropTypes.array.isRequired,
    future: PropTypes.array.isRequired
  }).isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
  contentEditable: PropTypes.bool.isRequired,
  updateValue: PropTypes.func.isRequired
};
