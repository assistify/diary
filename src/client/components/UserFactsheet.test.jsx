import React from 'react';
import renderer from 'react-test-renderer';
import UserFactsheet from './UserFactsheet';

it('renders correctly', () => {
  const tree = renderer
    .create(<UserFactsheet username="testuser" contentEditable={false} updateValue={() => {}}>
      <div className="sampleChild">sample Child should be rendered</div>
            </UserFactsheet>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
