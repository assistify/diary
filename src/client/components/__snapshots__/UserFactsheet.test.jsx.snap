// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<article
  className="media"
  style={Object {}}
>
  <figure
    className="media-left"
    style={Object {}}
  >
    <figure
      className="image avatar is-128x128"
      style={Object {}}
    >
      <img
        alt="testuser"
        onError={[Function]}
        src="/avatar/testuser"
      />
    </figure>
  </figure>
  <div
    className="factSheetInfo"
  >
    <h1
      className="title is-5"
    >
      <a
        className="c-user"
        href="/direct/testuser"
        rel="noopener noreferrer"
        target="_blank"
      >
        @testuser
      </a>
    </h1>
    <div
      className="sampleChild"
    >
      sample Child should be rendered
    </div>
  </div>
</article>
`;
